v2/processor.go

```diff
@@ -133,37 +196,38 @@ func (p *Processor) start(ctx context.Context) error {
 			if ctx.Err() != nil || maxMessages == 0 {
 				break
 			}
-			messages, err := p.receiver.ReceiveMessages(ctx, maxMessages, nil)
+			messages, err := receiver.ReceiveMessages(ctx, maxMessages, nil)
 			if err != nil {
-				return err
+				return fmt.Errorf("processor %s failed to receive messages: %w", receiverName, err)
 			}
-			log(ctx, fmt.Sprintf("received %d messages from processor loop", len(messages)))
-			processor.Metric.IncMessageReceived(float64(len(messages)))
+			log(ctx, fmt.Sprintf("processor %s received %d messages from processor loop", receiverName, len(messages)))
+			processor.Metric.IncMessageReceived(receiverName, float64(len(messages)))
 			for _, msg := range messages {
-				p.process(ctx, msg)
+				p.process(ctx, receiverName, msg)
 			}
 		case <-ctx.Done():
-			log(ctx, "context done, stop receiving")
+			log(ctx, fmt.Sprintf("context done, stop receiving from processor %s", receiverName))
 			break
 		}
 	}
-	log(ctx, "exiting processor")
-	return ctx.Err()
+	log(ctx, fmt.Sprintf("exiting processor %s", receiverName))
+	return fmt.Errorf("processor %s stopped: %w", receiverName, ctx.Err())
 }
 
-func (p *Processor) process(ctx context.Context, message *azservicebus.ReceivedMessage) {
+func (p *Processor) process(ctx context.Context, receiverName string, message *azservicebus.ReceivedMessage) {
+	receiver := p.receivers[receiverName].sbReceiver
 	p.concurrencyTokens <- struct{}{}
 	go func() {
 		msgContext, cancel := context.WithCancel(ctx)
 		// cancel messageContext when we get out of this goroutine
 		defer cancel()
 		defer func() {
 			<-p.concurrencyTokens
-			processor.Metric.IncMessageHandled(message)
+			processor.Metric.IncMessageHandled(receiverName, message)
 			processor.Metric.DecConcurrentMessageCount(message)
 		}()
 		processor.Metric.IncConcurrentMessageCount(message)
```

Comment:
nit: you could also add the receiver name to the concurrent message count metric.
that allows you to count all, or per receiver, and see the metric change when we failover for example.